src/rp2_common/pico_cyw43_driver/cyw43_bus_pio_spi.c

```diff
@@ -315,6 +316,12 @@ int cyw43_spi_transfer(cyw43_int_t *self, const uint8_t *tx, size_t tx_length, u
             tight_loop_contents(); // todo timeout
         }
         __compiler_memory_barrier();
+
+        while (!(dma_hw->intr & 1u << bus_data->dma_out)) {
```

Comment:
Will this have to re-calculate `1u << bus_data->dma_out` each time around the loop? Perhaps it's worth calculating it once, before the `while` loop?

Reply:
The compiler can hoist simple loop-invariant code like this, unless you have done something weird like mark `bus_data` as `volatile`.